src/main/java/com/shapesecurity/salvation/Parser.java

```diff
@@ -505,6 +498,7 @@ private void enforceMissingDirectiveValue(@Nonnull Token directiveNameToken) thr
         if (token.value.equalsIgnoreCase("'self'")) {
             return KeywordSource.Self;
         }
+        checkForUnquotedKeyword(token);
         if (token.value.matches("^" + Constants.schemePart + ":$")) {
```

Comment:
From here to the end of the function is the same as 438-460. Is there any way we can dedupe this as well? And we can probably even do the same for `'none'` and `'self'` above. If you want, I can take a crack at it.